frame/vamm/src/tests.rs

```diff
@@ -0,0 +1,181 @@
+		if System::block_number() > 0 {
+			Timestamp::on_finalize(System::block_number());
+			System::on_finalize(System::block_number());
+		}
+		System::set_block_number(System::block_number() + 1);
+		// Time is set in milliseconds, so at each block we increment the timestamp by 1000ms = 1s
+		let _ = Timestamp::set(Origin::none(), System::block_number() * 1000);
+		System::on_initialize(System::block_number());
+		Timestamp::on_initialize(System::block_number());
+	}
+}
+
+// ----------------------------------------------------------------------------------------------------
+//                                             Prop_compose
+// ----------------------------------------------------------------------------------------------------
+
+#[allow(dead_code)]
+const ZERO_RESERVE: Balance = Balance::MIN;
+
+#[allow(dead_code)]
+const MINIMUM_RESERVE: Balance = ZERO_RESERVE + 1;
+
+#[allow(dead_code)]
+const MAXIMUM_RESERVE: Balance = Balance::MAX;
+
+#[allow(dead_code)]
+const RUN_CASES: u32 = 100;
+
+// ----------------------------------------------------------------------------------------------------
+//                                             Create Vamm
+// ----------------------------------------------------------------------------------------------------
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	#[allow(clippy::disallowed_methods)]
+	fn create_vamm(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE) {
+		ExtBuilder::default().build().execute_with(|| {
+			let vamm_counter = Vamm::vamm_count();
+
+			let vamm_expected = VammState::<Balance, <MockRuntime as pallet::Config>::Timestamp> {
+					base_asset_reserves,
+					quote_asset_reserves,
+					peg_multiplier,
+					deprecated: Default::default(),
+			};
+
+			let vamm_created_ok = Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier);
+			let vamm_created_some = Vamm::get_vamm(vamm_created_ok.unwrap());
+
+			assert_ok!(vamm_created_ok);
+			assert_eq!(vamm_created_some, Some(vamm_expected));
+
+			assert_eq!(Vamm::vamm_count(), vamm_counter+1);
+		});
+	}
+}
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	fn create_vamm_succeeds(base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE) {
+		ExtBuilder::default().build().execute_with(|| {
+			assert_ok!(Vamm::create(base_asset_reserves, quote_asset_reserves, peg_multiplier));
+		});
+	}
+}
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	fn create_vamm_zero_base_asset_reserves_error(base_asset_reserves in ZERO_RESERVE..=ZERO_RESERVE, quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE, peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE) {
```

Comment:
you can remove base_asset_reserves in ZERO_RESERVE..=ZERO_RESERVE and just hardcode ZERO_RESERVE like so:

`assert_noop!(Vamm::create(ZERO_RESERVE, quote_asset_reserves, peg_multiplier), Error::<MockRuntime>::BaseAssetReserveIsZero);`

but that is up to you. The same applies to the below functions that check for the same thing with quote_asset_reserves and peg_multiplier

Reply:
Oh, yeah this way is definitely better, I changed to it :+1: